Wikipedia/UI-V5/WMFArticleContainerViewController.m

```diff
@@ -366,6 +379,66 @@ - (UIBarButtonItem*)languagesToolbarItem {
     return _languagesToolbarItem;
 }
 
+#pragma mark - Progress
+
+- (void)addProgressView {
+    [self.view addSubview:self.progressView];
+    [self.progressView mas_makeConstraints:^(MASConstraintMaker* make) {
+        make.top.equalTo(self.navigationController.navigationBar.mas_bottom).with.offset(0);
+        make.left.equalTo(self.progressView.superview.mas_left);
+        make.right.equalTo(self.progressView.superview.mas_right);
+        make.height.equalTo(@2.0);
+    }];
+}
+
+- (void)removeProgressView {
+    [self.progressView removeFromSuperview];
+}
+
+- (void)showProgressViewAnimated:(BOOL)animated {
+    self.progressView.progress = 0.0;
+
+    if (!animated) {
+        [self _showProgressView];
+        return;
+    }
+
+    [UIView animateWithDuration:0.25 animations:^{
+        [self _showProgressView];
+    } completion:^(BOOL finished) {
+    }];
+}
+
+- (void)_showProgressView {
+    self.progressView.alpha = 1.0;
+}
+
+- (void)hideProgressViewAnimated:(BOOL)animated {
+    //Don't remove the progress immediately, let the user see it then dismiss
+    dispatchOnMainQueueAfterDelayInSeconds(0.5, ^{
+        if (!animated) {
+            [self _hideProgressView];
+            return;
+        }
+
+        [UIView animateWithDuration:0.25 animations:^{
+            [self _hideProgressView];
+        } completion:nil];
+    });
+}
+
+- (void)_hideProgressView {
+    self.progressView.alpha = 0.0;
+}
+
+- (void)updateProgress:(CGFloat)progress animated:(BOOL)animated {
+    [self.progressView setProgress:progress animated:animated];
+}
+
+- (CGFloat)totalProgressWithArticleFetcherProgress:(CGFloat)progress {
+    return 0.75 * progress;
```

Comment:
Yeah - so far thats not been a great API - don't see many people really using that.